src/server/scripts/poller.js

```diff
@@ -108,16 +111,30 @@ export const checkSnapRepository = async (owner, name, last_polled_at) => {
 };
 
 
-// Request a build of a given snap repository (in LP) and update `polled_at` (in DB).
+/** Request a build of the corresponding snap repository (in LP) and
+ *  update `polled_at` (in DB).
+ *
+ * Return a `Promise` with the result of the operation.
+ */
 export const buildSnapRepository = async (owner, name) => {
-  const _now = new Date().getTime();
-  // XXX request LP build
-  // const repo_url = getGitHubRepoUrl(owner, name);
+  // TODO: annotate why the repository is being built (change on the main repo
+  // or in parts? which part ?).
+  const repositoryUrl = getGitHubRepoUrl(owner, name);
+  try {
+    const snap = await internalFindSnap(repositoryUrl);
+    await internalRequestSnapBuilds(snap, owner, name);
+    logger.info(`Requested builds of ${repositoryUrl}.`);
+  } catch (e) {
+    logger.error(`Failed to request builds of ${repositoryUrl}: ${e}.`);
+    return Promise.reject(e);
+  }
+
   return db.transaction(async (trx) => {
     const row = await db.model('Repository')
       .where({ owner, name })
       .fetch({ transacting: trx });
-    await row.save({ polled_at: _now }, { method: 'update', transacting: trx });
+    await row.save(
+      { polled_at: new Date() }, { method: 'update', transacting: trx });
```

Comment:
Thanks @bartaz, your points are spot on. I will work on a change to replace `poller_at` from DB with a Snap/Build fetch from LP. That should fix the accuracy of the mismatch checking and clean the design (keeping the app stateless).